ui/src/shared/components/FieldList.js

```diff
@@ -58,6 +64,49 @@ class FieldList extends Component {
     this.props.onFill(fill)
   }
 
+  handleToggleField = field => {
+    const {
+      query,
+      onToggleField,
+      addInitialField,
+      initialGroupByTime: time,
+      isKapacitorRule,
+    } = this.props
+    const {fields, groupBy} = query
+    const initialGroupBy = {...groupBy, time}
+
+    if (!_.size(fields)) {
+      return isKapacitorRule
+        ? onToggleField(field)
+        : addInitialField(field, initialGroupBy)
+    }
+
+    onToggleField(field)
+  }
+
+  handleApplyFuncs = fieldFunc => {
+    const {
+      query,
+      removeFuncs,
+      applyFuncsToField,
+      initialGroupByTime: time,
+    } = this.props
+    const {groupBy, fields} = query
+    const {funcs} = fieldFunc
```

Comment:
Is this valid anymore? I thought the schema changed

Reply:
@lukevmorris yes, it is.  the shape of the data in `FieldListItem` and `FieldList` is not the same as the shape of `fields` in the reducer or `queryConfig`